web/src/message_edit.ts

```diff
@@ -235,6 +236,10 @@ export function is_message_sent_by_my_bot(message: Message): boolean {
 }
 
 export function get_deletability(message: Message): boolean {
+    if (page_params.is_spectator) {
+        return false;
+    }
```

Comment:
Hmm, why do we need this? I think `user_has_permission_for_group_setting` is always false for spectators. So it makes me wonder if this is papering over a bug in some function wrapping that.